### PR TITLE
Improve receive docs

### PR DESCRIPTION
**Pull request by sverker**
Clarify message matching order; first all patterns then next message
in queue.

Rename "mailbox" and "mail queue" to message queue.

receive after limit is almost 50 days.

### CODE REVIEW

File: system/doc/reference_manual/expressions.xml [hunk 1]
```diff
@@ -432,15 +432,17 @@ receive
 end</pre>
     <p>
       Fetches a received message present in the message queue of
-      the process. The patterns <c>Pattern</c> are sequentially matched
-      against the first message in the message queue, then the second,
-      and so on. Messages in the message queue are
+      the process. The first message in the message queue is matched
+      sequentially against the patterns from top to bottom. If no
+      match was found the matching sequence is repeated for the second
```
Comment (rickard-green) [inline system/doc/reference_manual/expressions.xml:437]:
```suggestion
      match was found, the matching sequence is repeated for the second
```

File: system/doc/reference_manual/expressions.xml [hunk 2]
```diff
@@ -470,13 +472,15 @@ after
...
-    <p><c>ExprT</c> is to evaluate to an integer. The highest allowed
-      value is 16#FFFFFFFF, that is, the value must fit in 32 bits.
-      <c>receive..after</c> works exactly as <c>receive</c>, except
+    <p><c>receive..after</c> works exactly as <c>receive</c>, except
       that if no matching message has arrived within <c>ExprT</c>
       milliseconds, then <c>BodyT</c> is evaluated instead. The
       return value of <c>BodyT</c> then becomes the return value
-      of the <c>receive..after</c> expression.</p>
+      of the <c>receive..after</c> expression. <c>ExprT</c> is to
+      evaluate to an integer. The highest allowed value is
```
Comment (rickard-green) [inline system/doc/reference_manual/expressions.xml:480]:
```suggestion
      evaluate to an integer, or the atom <c>infinity</c>. If it evaluates to an integer, the highest allowed value is
```

**Review by rickard-green — APPROVED** [review]

File: system/doc/reference_manual/expressions.xml (hunk 2 above)
Comment (rickard-green) [reply system/doc/reference_manual/expressions.xml:480]: Hmm, that's weird. I think it is quite confusing to first claim that it should evaluate to an integer and then half a page below add to it. It should probably also describe the allowed range of integers, `i.e.` 0..4294967295`, not just mention the largest value allowed.

**Comment by sverker** [comment]
@rickard-green Added a paragraph about the "asynchronousness" of signal reception.

File: system/doc/reference_manual/expressions.xml [hunk 3]
```diff
@@ -432,15 +432,17 @@ receive
...
     <p>
       Fetches a received message present in the message queue of
-      the process. The patterns <c>Pattern</c> are sequentially matched
-      against the first message in the message queue, then the second,
-      and so on. Messages in the message queue are
+      the process. The first message in the message queue is matched
+      sequentially against the patterns from top to bottom. If no
+      match was found, the matching sequence is repeated for the second
+      message in the queue, and so on. Messages in the message queue are
       <seeguide marker="processes#message-queue-order">ordered in the
```
Comment (paulo-ferraz-oliveira) [inline system/doc/reference_manual/expressions.xml:439]: Would `delivered` be better than `ordered`, here?

Comment (sverker) [reply system/doc/reference_manual/expressions.xml:439]: I don't know. The text talks about the message queue as passive and the receive expression as active, how it matches and picks messages from the queue. So I think "delivered" is a bit confusing; from who to who?

Is it the "ordered in the order" phrasing that you don't like?

Comment (paulo-ferraz-oliveira) [reply system/doc/reference_manual/expressions.xml:439]: > Is it the "ordered in the order" phrasing that you don't like?

Yes, it's odd :) but I get your point.

Comment (paulo-ferraz-oliveira) [reply system/doc/reference_manual/expressions.xml:439]: Alternatives:
* "ordered as they were received"
* "kept in the order they were received"
* (longer form) "Messages are queued in the order they were received"

**Pull request merged**